### ISSUE TITLE
Devices

### ISSUE THREAD

**Issue by thebioengineer** [thebioengineer/camcorder#4]
Currently devices are limited to ones that can be read by `magick::image_read()`. To serve a wider variety of devices, we will need a converter just before to handle those images into one that `magick::image_read()` can handle, or change it so it can detect which method to use to read it...

things to look at: 
- help page for `magick::image_read()` - has demos for reading pdf, svg, etc. 

Changes to make:
- edit `image_resize` to make it more flexible

**Comment by gkaramanis** [comment]
Is this helpful at all (from the {magick} documentation):

> For reading svg or pdf it is recommended to use image_read_svg() and image_read_pdf() if the rsvg and pdftools R packages are available. These functions provide more rendering options (including rendering of literal svg) and better quality than built-in svg/pdf rendering delegates from imagemagick itself.

**Comment by Z3tt** [comment]
Yes, that's what I have used now. Works fine- I am not sure if we have to load the packages {pdftools} and {rsvg} or just need to import some functions? For now I have added the packages in the namespace via `import pdftools` and  `import rsvg`.

**Comment by Z3tt** [comment]
Tested some more file formats now:

✔️  bmp
❓   emf
❌  eps
✔️  jpeg
✔️  pdf
✔️  png
✔️  svg
✔️  tiff

I have no clue how to handle eps. For emf I couldn't even find a test file - who uses an "Enhanced Windows Meta File" format for storing ggplots? I think it both are uncommon output formats so would suggest to remove them from the list in `gg_record()`.

**Comment by Z3tt** [comment]
[This](https://davetang.org/muse/2019/02/22/importing-vector-images-into-r/) could be one approach to handle eps with the help of the {grImport} package but I am not sure if it's worth the effort.
